### ISSUE TITLE
Null pointer exception when no version supplied via an --init-config

### ISSUE THREAD

**Issue by cthulhuology** [aws-greengrass/aws-greengrass-nucleus#909]
If you omit the version of the Nucleus component in your --init-config file 

```json
{
  "system": {
    "certificateFilePath": "/greengrass/gg-config/certs/test123.pem",
    "privateKeyPath": "/greengrass/gg-config/certs/test123.pem",
    "rootCaPath": "/greengrass/gg-config/certs/AmazonRootCA1.pem",
    "thingName": "test123",
    "rootpath": "/greengrass/"
  },
  "services": {
    "aws.greengrass.Nucleus": {
      "componentType": "NUCLEUS",
      "configuration": {
        "awsRegion": "eu-west-1",
        "iotRoleAlias": "GreengrassV2TokenExchangeRoleAlias",
        "iotDataEndpoint": "a3a00z24dknd3b-ats.iot.eu-west-1.amazonaws.com",
        "iotCredEndpoint": "c1w19xtvvf60nl.credentials.iot.eu-west-1.amazonaws.com",
        "mqtt": {
          "port": 443
        },
   ...
}
```

And then you will get a null pointer exception during a deployment:

```
2021-04-06T13:32:23.607Z [INFO] (pool-2-thread-10) com.aws.greengrass.deployment.DeploymentService: Current deployment finished. {DeploymentId=f633f88d-1f4d-4d72-9fe5-24d84624d541, serviceName=DeploymentService, currentState=RUNNING}
2021-04-06T13:32:23.609Z [ERROR] (pool-2-thread-10) com.aws.greengrass.deployment.DeploymentService: Deployment task throws unknown exception. {DeploymentId=f633f88d-1f4d-4d72-9fe5-24d84624d541, serviceName=DeploymentService, currentState=RUNNING}
java.lang.NullPointerException
        at com.aws.greengrass.lifecyclemanager.GenericExternalService.isBootstrapRequired(GenericExternalService.java:235)
        at com.aws.greengrass.deployment.bootstrap.BootstrapManager.serviceBootstrapRequired(BootstrapManager.java:332)
        at com.aws.greengrass.deployment.bootstrap.BootstrapManager.lambda$isBootstrapRequired$0(BootstrapManager.java:131)
        at java.base/java.util.HashMap.forEach(HashMap.java:1336)
        at com.aws.greengrass.deployment.bootstrap.BootstrapManager.isBootstrapRequired(BootstrapManager.java:130)
        at com.aws.greengrass.deployment.activator.DeploymentActivatorFactory.getDeploymentActivator(DeploymentActivatorFactory.java:34)
        at com.aws.greengrass.deployment.DeploymentConfigMerger.mergeInNewConfig(DeploymentConfigMerger.java:69)
        at com.aws.greengrass.deployment.DefaultDeploymentTask.call(DefaultDeploymentTask.java:112)
        at com.aws.greengrass.deployment.DefaultDeploymentTask.call(DefaultDeploymentTask.java:36)
        at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
        at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
        at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
        at java.base/java.lang.Thread.run(Thread.java:834)
```

And the deployment will enter a fail state.  Nucleus knows its version however, and shouldn't depend on this input for the configuration.

**Comment by MikeDombo** [comment]
Hi Dave,
Thanks for letting us know. We'll discuss with the team what can be done, however, the Nucleus actually gets its version from the config when the config exists. So since you are providing a configuration, you must also provide the Nucleus version as we mention in the manual installation documentation: https://docs.aws.amazon.com/greengrass/v2/developerguide/manual-installation.html#run-greengrass-core-v2-installer-manual

**Comment by MikeDombo** [comment]
Looks like we have actually addressed it in #841. So that fix should be included in our next release.

**Comment by MikeDombo** [comment]
Since we have already addressed this bug I'm going to close this issue. Please reopen it if you need to discuss it any further.
